agenthub/codeact_agent/codeact_agent.py

```diff
@@ -210,6 +211,8 @@ def step(self, state: State) -> Action:
 
         try:
             response = self.llm.completion(**params)
+        except UserCancelledError as e:
+            raise e
         except Exception as e:
             logger.error(f'{e}')
             error_message = '{}: {}'.format(type(e).__name__, str(e).split('\n')[0])
```

Comment:
Oh, not in scope of this PR, but this doesn't look quite right. IMHO we shouldn't have needed to add this kind of exception in the agent, I don't think, it should have simply fallen down to the controller. But it won't because we catch everything and not re-raise.
One more consequence of this code here, is that now _every_ agent needs to handle ctrl+c, and that cannot be right...

Perhaps we can add a TODO: fix exception handling?

Reply:
Yeh, we need a separate issue for that.